ui/admin/app/routes/scopes/scope/roles/role/add-principals.js

```diff
@@ -66,11 +66,35 @@ export default class ScopesScopeRolesRoleAddPrincipalsRoute extends Route {
           }
         )
       : this.store.query('group', { scope_id: 'global', recursive: true });
+    //query authmethods from all the scopes
+    const oidcAuthMethods = scopeIDs?.length
+      ? await this.resourceFilterStore.queryBy(
+          'auth-method',
+          { scope_id: scopeIDs, type: 'oidc' },
+          { scope_id: 'global', recursive: true }
+        )
+      : await this.resourceFilterStore.queryBy(
+          'auth-method',
+          { type: 'oidc' },
+          { scope_id: 'global', recursive: true }
+        );
+    //extract oidc authMethod IDs
+    const authMethodIDs = oidcAuthMethods.map(({ id }) => id);
```

Comment:
Suggestion, not blocking comment. Since the ID's are extracted from oidc auth methods and we already have `oidcAuthMethods` I would souggest name this variable `oidcAuthMethodsIDs` to keep descriptive naming.

Reply:
done!